src/pkcs11/framework-pkcs15.c

```diff
@@ -3879,8 +3879,9 @@ pkcs15_pubkey_get_attribute(struct sc_pkcs11_session *session, void *object, CK_
 		break;
 	case CKA_TOKEN:
 	case CKA_SENSITIVE:
```

Comment:
I don't know why - but it is clearly against both the standard and the common sense (the whole idea of PKC is that you can freely share public keys).

And I definitely am not the one to know who and why wrote that code. I can only say that it looks like a bug.